cmd/apps/kuma_app.go

```diff
@@ -0,0 +1,117 @@
+
+	kuma.Flags().String("control-plane-mode", "standalone", "Kuma CP modes: one of standalone,zone,global")
+	kuma.Flags().Bool("auto-scale", false, "Enable Horizontal Pod Autoscaling (requires the Metrics Server)")
+
+	kuma.Flags().Bool("use-cni", false, "Use CNI instead of proxy init container")
+	kuma.Flags().Bool("ingress", false, "Deploy Ingress for cross cluster communication")
+
+	kuma.Flags().StringArray("set", []string{}, "Use custom flags or override existing flags \n(example --set ingress.enabled=false)")
+
+	kuma.RunE = func(command *cobra.Command, args []string) error {
+		kubeConfigPath, _ := command.Flags().GetString("kubeconfig")
+		if err := config.SetKubeconfig(kubeConfigPath); err != nil {
+			return err
+		}
+
+		arch := k8s.GetNodeArchitecture()
+		fmt.Printf("Node architecture: %q\n", arch)
+		if suffix := getValuesSuffix(arch); suffix == "-armhf" {
+			return fmt.Errorf(`Kuma is currently not supported on armhf architectures`)
+		}
+
+		namespace, _ := command.Flags().GetString("namespace")
+		controlPlaneMode, _ := command.Flags().GetString("control-plane-mode")
+		autoScale, _ := command.Flags().GetBool("auto-scale")
+		useCNI, _ := command.Flags().GetBool("use-cni")
+		ingress, _ := command.Flags().GetBool("ingress")
+		customFlags, _ := command.Flags().GetStringArray("set")
+
+		overrides := map[string]string{}
+
+		// control-plane-mode  standalone,zone,global
+		if controlPlaneMode == "standalone" || controlPlaneMode == "zone" || controlPlaneMode == "global" {
+			overrides["controlPlane.mode"] = controlPlaneMode
+		} else {
+			return fmt.Errorf(`Kuma's control-plane mode must be one of global, zone or standalone.`)
+		}
+
+		if autoScale {
+			overrides["controlPlane.autoscaling.enabled"] = "true"
+		}
+
+		if useCNI {
+			overrides["cni.enabled"] = "true"
+		}
+
+		if ingress {
+			overrides["ingress.enabled"] = "true"
+		}
+
+		if err := config.MergeFlags(overrides, customFlags); err != nil {
+			return err
+		}
+
+		kumaOptions := types.DefaultInstallOptions().
+			WithNamespace(namespace).
+			WithHelmRepo("kuma/kuma").
+			WithHelmURL("https://kumahq.github.io/charts").
+			WithOverrides(overrides).
+			WithKubeconfigPath(kubeConfigPath)
+
+		_, err := apps.MakeInstallChart(kumaOptions)
+		if err != nil {
+			return err
+		}
+
+		println(kumaInstallMsg)
+		return nil
+	}
+
+	return kuma
+}
+
+const KumaInfoMsg = `
+Kuma has been installed, you can access the control-plane via either the GUI, kubectl, the HTTP API, or the CLI:
+
+# You can use kubectl without forwarding the port:
+kubectl get meshes
+
+# Forward the port with:
+kubectl port-forward svc/kuma-control-plane -n kuma-system 5681:5681
+
+## You can access the GUI on: http://127.0.0.1:5681/gui
+
+## You can access the API on: http://127.0.0.1:5681
+
+## You can use kumactl: kumactl get meshes
+
+# Find out more on the project homepage:
+# https://kuma.io/docs/1.3.1/documentation/overview/#kubernetes-mode
```

Comment:
Is there a link that takes you to latest? I only ask because you are not installing 1.3.1, so this message will get stale over time.

Reply:
If not, perhaps we can link to higher up?

Reply:
Sadly no, all the docs and download links on kuma's site must have a strict version link init. :disappointed: